modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -634,32 +634,42 @@ public void requestFocusVisible(Node node) {
 
             @Override
             public StyleableProperty<TransitionDefinition[]> getTransitionProperty(Node node) {
-                if (node.transitions == null) {
-                    node.transitions = node.new TransitionDefinitionCollection();
+                var definitions = node.miscProperties != null ? node.miscProperties.transitionDefinitions : null;
+                if (definitions == null) {
+                    definitions = node.new TransitionDefinitionCollection();
```

Comment:
TransitionDefinitionCollection can be a static class, can it?

Reply:
Yes, good catch!